### DISCUSSION TITLE
How to highlight trailing whitespace?

### DISCUSSION THREAD

**Discussion by azzamsa** [LazyVim/LazyVim#2219] [Q&A] [answered]
Hi.

I can highlight trailing whitespace using the code below, but it also applies to `nvimdev/dashboard-nvim` buffer.
Any better solutions?

```lua

-- Highlight trailing whitespace with a red background
vim.api.nvim_exec(
	[[
  highlight TrailingWhitespace guibg=red
  match TrailingWhitespace /\s\+$/
]],
	false
)
```

**Comment by dpetka2001** [comment]
Using an autocmd might help
```lua
-- ~/.config/nvim/lua/config/autocmds.lua
vim.api.nvim_create_autocmd("FileType", {
  group = vim.api.nvim_create_augroup("hi_trail_whsp", { clear = true }),
  pattern = {
    "dashboard",
  },
  callback = function()
    vim.api.nvim_exec(
      [[highlight TrailingWhitespace guibg=red
  match TrailingWhitespace /\s\+$/]],
      false
    )
  end,
  desc = "Highlight Trailing Whitespace",
})
```
Test it to see if it works.

**Reply by azzamsa** [reply]
Hi, Thanks for the code. Unfortunately, it doesn't work. It should highlight the trailing line below:

![image](https://github.com/LazyVim/LazyVim/assets/17734314/a81bdf54-4c87-4529-b69c-bea9f781f95f)

**Reply by dpetka2001** [reply]
Try in `~/.config/nvim/lua/config/autocmds.lua` the following
```lua
vim.cmd([[
  autocmd BufReadPre * hi TrailingWhitespace guibg=red|match TrailingWhitespace /\s\+$/
]])
```
Hopefully it does what you want.

**Reply by azzamsa** [reply, marked as answer]
It worked like a charm! 🎖️

It works only on buffers but not dashboard buffer.
By the way, which line of the code does not highlight the dashboard buffer?

![image](https://github.com/LazyVim/LazyVim/assets/17734314/727dd072-b1af-4e8a-967c-7847b365d8c0)

**Reply by dpetka2001** [reply]
This just creates an auto command on the event `BufReadPre`, which happens just before a buffer gets read. Since the dashboard is not a buffer, it won't apply to it.
